platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/cache/VirtualNodeCache.java

```diff
@@ -1394,7 +1518,7 @@ private void serializePathToDirtyHashIndex(
             assert mutation.version <= this.fastCopyVersion.get()
                     : "Trying to serialize pathToDirtyInternalIndex with a version ahead";
             out.writeLong(mutation.version);
-            out.writeBoolean(mutation.isDeleted());
+            out.writeByte(mutation.getFlags());
```

Comment:
This looks like a serialized data format change. Is this backward compatible? Does this require any migration? Is the data on disk considered "public" to require a formal design review?

Reply:
Yes, this is compatible, I double checked. Booleans are written as bytes. No migration is needed.

Previously, the byte was either 1 (deleted) or 0 (not deleted). Now this byte is read as "flags", and then "new" and "deleted" are extracted from it:

```java
            final byte flags = in.readByte();
            final boolean isNew = Mutation.getFlag(flags, Mutation.FLAG_BIT_NEW);
            final boolean isDeleted = Mutation.getFlag(flags, Mutation.FLAG_BIT_DELETED);
```

If the byte was 0, then both "new" and "deleted" are false. If the byte was "1", then "new" is false, and "deleted" is true, as expected. `FLAG_BIT_DELETED` above is 0.